lib/openssl.c

```diff
@@ -1220,7 +1220,7 @@ static int verify_cert_chain(X509_STORE *store, X509 *cert, STACK_OF(X509) * cha
     X509_STORE_CTX *verify_ctx;
     int ret;
 
-    assert(server_name != NULL && "ptls_set_server_name MUST be called");
+    //assert(server_name != NULL && "ptls_set_server_name MUST be called");
```

Comment:
Maybe:
```suggestion
    if (!is_server)
        assert(server_name != NULL && "ptls_set_server_name MUST be called");
```

Client has to specify the server's name, otherwise it cannot verify that the server it is talking to is the one that the client intended to.

There might be exceptional cases (e.g., when Raw Public Keys is being used) that it might be possible to skip server-name validation though. Is that your problem?

Reply:
oh i guess i shouldn't have commented out the assert but essentially my point was to propose allowing verification by either 1) a provided root certificate chain OR 2) SNI. but i just realized there's no way at the moment to differentiate between the default root cert store and a provided chain.

another interesting option might be pinning to the server's public key? in theory it provides a level of protection beyond SNI. i don't require it but just to mention it.

that said, this was essentially an unnecessary addendum to the problem i thought this PR was correcting so i'm just going to close it!